foundation/foundation-upgrade/src/main/java/com/boozallen/aissemble/upgrade/migration/v1_8_0/ExtensionsSecurityProjectsMigration.java

```diff
@@ -0,0 +1,92 @@
+package com.boozallen.aissemble.upgrade.migration.v1_8_0;
+
+/*-
+ * #%L
+ * aiSSEMBLE::Foundation::Upgrade
+ * %%
+ * Copyright (C) 2021 Booz Allen
+ * %%
+ * This software package is licensed under the Booz Allen Public License. All Rights Reserved.
+ * #L%
+ */
+
+import com.boozallen.aissemble.upgrade.migration.AbstractAissembleMigration;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.io.BufferedReader;
+import java.io.File;
+import java.io.FileReader;
+import java.util.Map;
+
+import static org.technologybrewery.baton.util.FileUtils.replaceLiteralInFile;
+
+/**
+ Baton migration class to verify whether a migrations are required for any outdated PDP and/or authzforce modules which
+ are referencing AIOPS instead of aiSSEMBLE.
+ **/
+
+public class ExtensionsSecurityProjectsMigration extends AbstractAissembleMigration {
+
+    public static final Logger logger = LoggerFactory.getLogger(ExtensionsSecurityProjectsMigration.class);
+    public static final Map<String, String> EXTENSIONS_SECURITY_REFERENCE_PACKAGE_MAP = Map.of(
+            // aissemble/security/authorization/
+            "AiopsKeycloakSecureTokenServiceClient","AissembleKeycloakSecureTokenServiceClient",
+            "AiopsKeyStore","AissembleKeyStore",
+            "AiopsSecureTokenServiceClient","AissembleSecureTokenServiceClient",
+            "AiopsSimpleSecureTokenServiceClient","AissembleSimpleSecureTokenServiceClient",
+
+            // aissemble/security/authorization/policy/
+            "AiopsAttribute","AissembleAttribute",
+            "AiopsAttributePoint","AissembleAttributePoint",
+            "AiopsAttributeProvider","AissembleAttributeProvider",
+            "AiopsAttributeUtils","AissembleAttributeUtils",
+
+            // aissemble/security/authorization/exception
+            "AiopsSecurityException", "AissembleSecurityException"
+
+    );
+
+    /**
+     * Function to check whether the migration is necessary.
+     * @param file file to check
+     * @return shouldExecute - whether the migration is necessary.
+     */
+    @Override
+    protected boolean shouldExecuteOnFile(File file) {
```

Comment:
a: shouldExecuteOnFile and performMigration seem to be the same as AiopsReferenceJavaMigration. Could they be combined or common functions abstracted?

Reply:
Due to the nature of AiopsReferenceJavaMigration being more of a dynamic plug an play migration script for imports and ExtensionsSecurityProjectsMigration being specific to the authzforce and PDP modules, I split them up for ease of review and testing.